ArduPlane/ArduPlane.cpp

```diff
@@ -908,6 +908,50 @@ bool Plane::set_land_descent_rate(float descent_rate)
 #endif
     return false;
 }
+
+// allow scripte to find access prev_WP_loc
+bool Plane::get_previous_location(Location &previous_location)
+{
+    switch (control_mode->mode_number()) {
+    case Mode::Number::RTL:
+    case Mode::Number::AVOID_ADSB:
+    case Mode::Number::GUIDED:
+    case Mode::Number::AUTO:
+    case Mode::Number::LOITER:
+    case Mode::Number::TAKEOFF:
+#if HAL_QUADPLANE_ENABLED
+    case Mode::Number::QLOITER:
+    case Mode::Number::QLAND:
+    case Mode::Number::QRTL:
+#endif
```

Comment:
I think this should be the same modes as allow setting it. The problem is that if the mode does not use that location it really could be anywhere so you might end up navigating to somewhere new rather than somewhere you have already been.

Reply:
TBH, I'm not really a fan of this method at all. We have a binding for current destination. You could easly use that in lua for the same result. 

If we were to add a binding for the `same_loc_as` location method:
```
local previous_location

local function update()

    local target = vehicle:get_target_location()
    if (target and (not previous_location or not previous_location:same_loc_as(target)) then
        previous_location = target
    end

   ....
end
```
If we dont add a binding for `same_loc_as` then you can add a little helper:
```
local function same_loc_as(A, B)
    if (A:lat() ~= B:lat()) or (A:lon() ~= B:lon()) then
        return false
    end
    return (A:alt() == B:alt()) and (A:get_alt_frame() == B:get_alt_frame())
end
```

Reply:
> I think this should be the same modes as allow setting it. The problem is that if the mode does not use that location it really could be anywhere so you might end up navigating to somewhere new rather than somewhere you have already been.

prev_WP_loc is set in Plane::update_GPS_10Hz as well as a number of other places like do_RTL(), do_takeoff(), verify_nav_wp() set_next_wp(), MAV_CMD_GUIDED_CHANGE_HEADING, ModeCruise::navigate()

My suggestion is to allow it to be used "as required" in Lua regardless of where its set. Why restrict it?

Reply:
> TBH, I'm not really a fan of this method at all. We have a binding for current destination. You could easly use that in lua for the same result.
> 
> If we were to add a binding for the `same_loc_as` location method:
> 
> ```
> local previous_location
> 
> local function update()
> 
>     local target = vehicle:get_target_location()
>     if (target and (not previous_location or not previous_location:same_loc_as(target)) then
>         previous_location = target
>     end
> 
>    ....
> end
> ```
> 
> If we dont add a binding for `same_loc_as` then you can add a little helper:
> 
> ```
> local function same_loc_as(A, B)
>     if (A:lat() ~= B:lat()) or (A:lon() ~= B:lon()) then
>         return false
>     end
>     return (A:alt() == B:alt()) and (A:get_alt_frame() == B:get_alt_frame())
> end
> ```

I don't understand this

Reply:
> TBH, I'm not really a fan of this method at all. We have a binding for current destination. You could easly use that in lua for the same result.

As far as I can see prev_WP_loc may never have been a previous "target_location" based on all the places it is set in the code.

prev_WP_loc is set in Plane::update_GPS_10Hz as well as a number of other places like do_RTL(), do_takeoff(), verify_nav_wp() set_next_wp(), MAV_CMD_GUIDED_CHANGE_HEADING, ModeCruise::navigate() ...

Reply:
> TBH, I'm not really a fan of this method at all. We have a binding for current destination. You could easly use that in lua for the same result.
> 
> If we were to add a binding for the `same_loc_as` location method:
> 
> ```
> local previous_location
> 
> local function update()
> 
>     local target = vehicle:get_target_location()
>     if (target and (not previous_location or not previous_location:same_loc_as(target)) then
>         previous_location = target
>     end
> 
>    ....
> end
> ```
> 
> If we dont add a binding for `same_loc_as` then you can add a little helper:
> 
> ```
> local function same_loc_as(A, B)
>     if (A:lat() ~= B:lat()) or (A:lon() ~= B:lon()) then
>         return false
>     end
>     return (A:alt() == B:alt()) and (A:get_alt_frame() == B:get_alt_frame())
> end
> ```

I've thought about this a lot and I think the idea of duplicating code that already exists in c++ in Lua is a bad idea. In principle - if the functionality exists in c++ the correct solution is to create a binding.

Reply:
I got this to work, example script updated.